calico_node/allocateipip/allocate_ipip_addr.go

```diff
@@ -116,11 +118,27 @@ func assignHostTunnelAddr(c *client.Client, nodename string, ipipCidrs []net.IPN
 		IPv4Pools: ipipCidrs,
 	}
 
-	if ipv4Addrs, _, err := c.IPAM().AutoAssign(args); err != nil {
-		log.WithError(err).Fatal("Unable to autoassign an address for IPIP")
-	} else if len(ipv4Addrs) == 0 {
+	var ipv4Addrs []net.IP
+
+	// Retry loop around AutoAssign to recover from a distributed race in case
+	// all nodes in a large cluster respond to a change in /pools simultaneously.
+	for a := 3; a > 0; a-- {
+		var err error
+		ipv4Addrs, _, err = c.IPAM().AutoAssign(args)
+
+		if err != nil {
+			log.WithError(err).Error("Unable to autoassign an address for IPIP:")
```

Comment:
formatting nit: I think you can drop the trailing `:` here - logrus will handle formatting the WithError properly.